docs/src/main/asciidoc/security-keycloak-authorization.adoc

```diff
@@ -1,39 +1,65 @@
-:keywords: sso oidc security keycloak
-:topics: security,authentication,authorization,keycloak,sso,oidc
-:extensions: io.quarkus:quarkus-oidc,io.quarkus:quarkus-keycloak-authorization
+:keywords: sso, oidc, security, keycloak
+:topics: security, authentication, authorization, keycloak, sso, oidc
+:extensions: io.quarkus:quarkus-oidc, io.quarkus:quarkus-keycloak-authorization
 
-Learn how to enable bearer token authorization in your Quarkus application using link:https://www.keycloak.org/docs/latest/authorization_services/index.html[Keycloak Authorization Services] for secure access to protected resources.
+Learn how to enable bearer token authorization in your Quarkus application by using link:https://www.keycloak.org/docs/latest/authorization_services/index.html[Keycloak Authorization Services] for secure access to protected resources.
 
-The `quarkus-keycloak-authorization` extension relies on `quarkus-oidc`.
-It includes a policy enforcer that regulates access to secured resources.
-Access is governed by permissions set in Keycloak.
-Currently, this extension is compatible solely with Quarkus xref:security-oidc-bearer-token-authentication.adoc[OIDC service applications].
+== Overview
 
-It provides a flexible and dynamic authorization capability based on Resource-Based Access Control.
+The `quarkus-keycloak-authorization` extension builds on the `quarkus-oidc` extension to offer advanced authorization capabilities. It includes a policy enforcer that dynamically regulates access to secured resources. Access is governed by permissions defined in Keycloak, supporting flexible and dynamic Resource-Based Access Control (RBAC).
 
-Rather than explicitly enforcing access through specific mechanisms such as role-based access control (RBAC), `quarkus-keycloak-authorization` determines request permissions based on resource attributes such as name, identifier, or Uniform Resource Identifier (URI).
-This process involves sending a `quarkus-oidc`-verified bearer access token to Keycloak Authorization Services for an authorization decision.
+Use this extension if:
 
-Use `quarkus-keycloak-authorization` only if you work with Keycloak and have Keycloak Authorization Services enabled to make authorization decisions.
-Use `quarkus-oidc` if you do not work with Keycloak or work with Keycloak but do not have its Keycloak Authorization Services enabled to make authorization decisions.
+- You are using Keycloak.
+- Keycloak Authorization Services is enabled in your environment to handle authorization decisions.
 
-By shifting authorization responsibilities outside your application, you enhance security through various access control methods while eliminating the need for frequent re-deployments whenever security needs evolve.
-In this case, Keycloak acts as a centralized authorization hub, managing your protected resources and their corresponding permissions effectively.
+For other cases, consider using the simpler `quarkus-oidc` extension instead.
```

Comment:
I'm not sure this sentence works, `for other cases` - which cases  and `simpler` also does not seem work...

May be we should only keep the original `Use 'quarkus-keycloak-authorization' only if you work with Keycloak and have Keycloak Authorization Services enabled to make authorization decisions.`

Reply:
Updated. Thanks.